Python/Product/TestAdapter/TestContainerDiscovererProject.cs

```diff
@@ -238,9 +241,9 @@ class TestContainerDiscovererProject : ITestContainerDiscoverer, IDisposable {
             _packageManagerEventSink.WatchPackageManagers(pyProj.GetInterpreterFactory());
 
             var projInfo = new ProjectInfo(pyProj);
-            _projectMap[projInfo.ProjectHome] = projInfo;
             var files = FilteredTestOrSettingsFiles(vsProject);
             UpdateContainersAndListeners(files, projInfo, isAdd: true);
+            _projectMap[projInfo.ProjectHome] = projInfo;
```

Comment:
Delay adding the new projectInfo collection of testconatiners until we finish populating it.